File: JavaScript/8js 1/invert.js
/*
|--------------------------------------------------------------------------
| Invert values
|--------------------------------------------------------------------------
|
*/

const invert = array => array.map(num => -num);


function invert(array) {
   return array.map(x => x === 0 ? x : -x);
}


const invert = a => a.map(n => ~~-n)


function invert(array) {
   return array.map(n => n && -n)
}


function invert(array) {
   return array.map(e => e * -1);
}


// function invert(array) {
//    let hasil = [];
//    for (let i in array) {
//       array[i] > 0 ? array[i] = array[i] * -1 : array[i] -= array[i] * 2;
//       hasil.push(array[i]);
//    }
//    return hasil;
// }


console.log(invert([1, 2, 3, 4, 5]), [-1, -2, -3, -4, -5]);
console.log(invert([1, -2, 3, -4, 5]), [-1, 2, -3, 4, -5]);
console.log(invert([]), []);
console.log(invert([0]), [-0]);

File: Python/8py 2/invert.py
"""
|--------------------------------------------------------------------------
| Invert values
|--------------------------------------------------------------------------
|
"""

def invert(lst):
   return list(map(lambda x: x * -1, lst))


def invert(lst):
   res=[]
   for num in lst:
      res.append(num*-1)
   return res


def invert(lst):
   return [-x for x in lst]


def invert(lst):
   return [] if lst == [] else [i*(-1) for i in lst]


print(invert([1,2,3,4,5]),[-1,-2,-3,-4,-5])
print(invert([1,-2,3,-4,5]), [-1,2,-3,4,-5])
print(invert([]), [])